src/org/traccar/protocol/GranitProtocolDecoder.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2016 Anton Tananaev (anton.tananaev@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.buffer.ChannelBuffers;
+import org.jboss.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.helper.BitUtil;
+import org.traccar.helper.Checksum;
+import org.traccar.model.Position;
+
+import java.net.SocketAddress;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class GranitProtocolDecoder extends BaseProtocolDecoder {
+
+    private static final int HEADER_LENGTH = 6;
+
+    public GranitProtocolDecoder(GranitProtocol protocol) {
+        super(protocol);
+    }
+
+    public static void appendChecksum(ChannelBuffer buffer, int length) {
+        buffer.writeByte(0x2a); // asterisk
```

Comment:
You can replace codes with actual symbols. Like `'*'` for asterisk. Same for end line symbols.